### DISCUSSION TITLE
PubSub Feeds Over XMPP

### DISCUSSION THREAD

**Discussion by sjehuda** [emacs-elfeed/elfeed#585] [General]
* Previous title: XMPP (RFC 6120), PubSub (XEP-0060) and Atom Over XMPP (RFC 4287)
* Original post: https://github.com/emacs-elfeed/elfeed/issues/535

More detailed information: https://portal.mozz.us/gemini/woodpeckersnest.space/~schapps/journal/2024-12-09-atom-over-xmpp-for-news-readers.gmi

If you are not famiiliar with XMPP; XMPP (previously Jabber) is a
messaging system with open standards. XMPP is extensible via XEP.

I will discuss about XEP-0060: Publish-Subscribe.

Almost a couple of decades ago, a discussion[2] arose about
incorporating Atom (RFC 4287) with XMPP PubSub, which is really easy
because both are XML data.

One of the major benefits of utilizing PubSub is native Push
Notifications (yes, that technique was much before HTML5) which do not
require HTTP pulling.

That would be a great feature to have photon to utilize XMPP PubSub
natively!

See a proof of concept[3] I have made to parse PubSub nodes as Atom
Syndication Format.

[1]: https://greasyfork.org/en/scripts/465932-newspaper-native-web-feed-reader
[2]: https://datatracker.ietf.org/meeting/66/materials/slides-66-atompub-1.pdf
[3]: https://codeberg.org/poezio/slixmpp/issues/3526

Cheers,
Schimon
If you are not famiiliar with XMPP; XMPP (previously Jabber) is a
messaging system with open standards. XMPP is extensible via XEP.

I will discuss about XEP-0060: Publish-Subscribe.

Almost a couple of decades ago, a discussion[1] arose about
incorporating Atom (RFC 4287) with XMPP PubSub, which is really easy
because both are XML data.

One of the major benefits of utilizing PubSub is native Push
Notifications (yes, that technique was much before HTML5) which do not
require HTTP pulling.

That would be a great feature to have photon to utilize XMPP PubSub
natively!

See a proof of concept[2] I have made to parse PubSub nodes as Atom
Syndication Format.

[1]: https://datatracker.ietf.org/meeting/66/materials/slides-66-atompub-1.pdf
[2]: https://codeberg.org/poezio/slixmpp/issues/3526

Further more, I have made an HTTP gateway (Rivista XJP) which returns
XMPP PubSub nodes as Atom Syndication Format

- Demonstration  : https://video.xmpp-it.net/w/vNqcMooy3pqRAZ8Yb8grr1
- Project source : https://git.xmpp-it.net/sch/Rivista

Cheers,

Schimon

**Comment by sjehuda** [comment]
This was recently implemented in project RSS Guard.

https://github.com/martinrotter/rssguard/issues/1548

Interestingly, the developer has further extended it to messages, even though I only advised to support PubSub and PEP, which should suffice.

He has done a remarkable work, to an extent which is good to study from.

**Comment by sjehuda** [comment]
This project might be relevant for XMPP references.

http://emacs-jabber.sourceforge.net

**Comment by minad** [comment]
Converting to discussion, since this proposal deviates quite from Elfeed's current scope.

**Reply by sjehuda** [reply]
Unless you ignore a single protocol.

To a syndication project, which I maintain, I created a module to handle multiple protocols.

https://git.xmpp-it.net/sch/Slixfeed/src/branch/master/src/slixfeed/utility/download.py

Data is realized equally; so, all that is required is to recognize format and parse it appropriately.

**Reply by minad** [reply]
For my taste you are promoting this too aggressively. If you want to see this implemented, you can provide code and we will consider it. Otherwise it is up to the maintainers and Elfeed users to see what they really need. There are more relevant matters in this project which need attention.

**Reply by sjehuda** [reply]
Fair.

Yes. I agree, I am promoting it in an unusual fashion.
